winml/adapter/winml_adapter_session.cpp

```diff
@@ -251,4 +251,21 @@ ORT_API_STATUS_IMPL(winmla::SessionGetNumberOfIntraOpThreads, _In_ OrtSession* s
   *num_threads = session_options.intra_op_param.thread_pool_size;
   return nullptr;
   API_IMPL_END
+}
+
+ORT_API_STATUS_IMPL(winmla::SessionGetNamedDimensionsOverrides, _In_ OrtSession* session, _Out_ winrt::Windows::Foundation::Collections::IMapView<winrt::hstring, uint32_t>& named_dimension_overrides) {
+  API_IMPL_BEGIN
+  auto inference_session = reinterpret_cast<::onnxruntime::InferenceSession*>(session);
+  auto session_options = inference_session->GetSessionOptions();
+  winrt::Windows::Foundation::Collections::IMap<winrt::hstring, uint32_t> override_map = winrt::single_threaded_map<winrt::hstring, uint32_t>();
```

Comment:
Does `IMapView` increment some reference counter the IMap? Will the data in `override_map` keep existing after the function returns?

Reply:
Asking because a view of `override_map` is returned from the function, but `override_map` is destroyed once the function returns.

Reply:
you know I assumed it did, but I can't find any explicit documentation right now. That being said, I utilize the API in the api test and it returns valid data, so it must reference count it

Reply:
Yeah, I saw that, but could also be use after free